src/devicecontroller/DefaultDeviceController.ts

```diff
@@ -811,24 +837,34 @@ export default class DefaultDeviceController
 
     // This function is called from `CleanStoppedSessionTask` using the
     // session state, which does not allow us to clean up any associated 'ended'
-    // callbacks in advance. Look here to see if we have any to clean up.
+    // or 'mute'/'unmute' callbacks in advance.
+    // Look here to see if we have any to clean up.
     for (const kind in this.activeDevices) {
       const activeDevice = this.activeDevices[kind];
       if (activeDevice?.stream !== mediaStreamToRelease) {
         continue;
       }
-      if (activeDevice.endedCallback) {
-        tracksToStop[0].removeEventListener('ended', activeDevice.endedCallback);
-        delete activeDevice.endedCallback;
-      }
-      delete this.activeDevices[kind];
 
-      if (
-        kind === 'video' &&
-        this.boundAudioVideoController?.videoTileController.hasStartedLocalVideoTile()
-      ) {
-        this.boundAudioVideoController.videoTileController.stopLocalVideoTile();
+      switch (kind) {
+        case 'audio': {
+          for (const track of mediaStreamToRelease.getAudioTracks()) {
+            track.removeEventListener('mute', activeDevice.trackMuteCallback);
```

Comment:
I see the `ended` event callback `removeEventListener` removed from above so checking if it should added here?
This shows removed.
```
tracksToStop[0].removeEventListener('ended', activeDevice.endedCallback);
```

Reply:
I think I found this to be unnecessary — this function is part of a stack that also does some cleanup work — but enough hours have passed that I'm going to have to re-check.